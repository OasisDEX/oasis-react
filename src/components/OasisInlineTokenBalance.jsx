import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import { formatAmount } from '../utils/tokens/pair';
import OasisLoadingIndicator from './OasisLoadingIndicator';


const propTypes = PropTypes && {
  balance: PropTypes.string,
  inWei: PropTypes.bool,
  token: PropTypes.string
};
const defaultProps = {};


class OasisInlineTokenBalance extends PureComponent {
  render() {
    const { balance, token, inWei } = this.props;
    if ((!!balance) && token) {
      return (
        <span>
          {formatAmount(balance, inWei)} <b>{token}</b>
        </span>
      );
    } else {
      return (<OasisLoadingIndicator/>);
    }
  }
}

OasisInlineTokenBalance.displayName = 'OasisInlineTokenBalance';
OasisInlineTokenBalance.propTypes = propTypes;
OasisInlineTokenBalance.defaultProps = defaultProps;
export default OasisInlineTokenBalance;
