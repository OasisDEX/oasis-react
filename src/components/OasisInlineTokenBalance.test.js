/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisInlineTokenBalance from './OasisInlineTokenBalance';


describe('(Component) OasisInlineTokenBalance', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <OasisInlineTokenBalance {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
