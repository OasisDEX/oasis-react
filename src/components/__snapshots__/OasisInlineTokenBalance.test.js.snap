// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisInlineTokenBalance should render 1`] = `<OasisLoadingIndicator />`;
